ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java

```diff
@@ -323,13 +323,16 @@ private static void handleExplicitKeyField(
       final Object rowKeyValue = values.get(SchemaUtil.ROWKEY_NAME);
 
       if (keyValue != null ^ rowKeyValue != null) {
-        values.putIfAbsent(key, rowKeyValue);
-        values.putIfAbsent(SchemaUtil.ROWKEY_NAME, keyValue);
-      } else if (!Objects.equals(keyValue, rowKeyValue)) {
-        throw new KsqlException(
-            String.format(
-                "Expected ROWKEY and %s to match but got %s and %s respectively.",
-                key, rowKeyValue, keyValue));
+        if (keyValue == null) {
+          values.putIfAbsent(key, rowKeyValue);
```

Comment:
with the guarding `if` block, this can now just be:

```suggestion
          values.put(key, rowKeyValue);
```

Right? 

Likewise for the one below.